cpp/src/barretenberg/honk/composer/composer_helper/standard_honk_composer_helper.cpp

```diff
@@ -129,11 +98,34 @@ std::shared_ptr<StandardHonkComposerHelper::VerificationKey> StandardHonkCompose
     if (verification_key) {
```

Comment:
are this checks due to potential multithreading? if so, can't we use something like `std::call_once`? otherwise, shouldn't all keys be initialised at some point in the beginning and then we're certain they're not null? maybe i'm misunderstanding the codebase

Reply:
I dont think it has to do with multithreading. Checks like this and the ones below seem to be left over from very old code and they're currently used in various tests but I don't think they're ever needed in production. I'd like to remove them but the scope is a bit more than I'd like for this PR.